drivers/hmis/spec/models/hmis/hmis_data_cleanup_util_spec.rb

```diff
@@ -129,6 +129,138 @@ def expect_leaves_non_hmis_data_alone(&block)
+      e2.update_columns(disabling_condition: nil)
+      e3.update_columns(disabling_condition: 1)
+      e4.update_columns(disabling_condition: 99)
+      GrdaWarehouse::Hud::Enrollment.where.not(data_source: hmis_ds).update_all(disabling_condition: nil)
+    end
+
+    it 'sets DisablingCondition from nil to 99, leaving other DisablingConditions unchanged' do
+      expect do
+        HmisDataCleanup::Util.fix_disabling_condition_nils!
+        [e1, e2, e3, e4].map(&:reload)
+      end.to change(e1, :disabling_condition).from(nil).to(99).
+        and change(e2, :disabling_condition).from(nil).to(99).
+        and not_change(e3, :disabling_condition).
+        and not_change(e4, :disabling_condition)
+    end
+
+    it 'does not make changes to non-hmis data' do
+      expect_leaves_non_hmis_data_alone do
+        HmisDataCleanup::Util.fix_disabling_condition_nils!
+      end
+    end
+  end
+
+  context 'clients with badly formatted race/gender data' do
+    let!(:race_fields) { HudUtility2024.races.keys.excluding('RaceNone') }
+    let!(:gender_fields) { HudUtility2024.gender_fields.excluding(:GenderNone) }
+
+    let!(:c1_with_race_and_gender) do
+      create(
+        :hmis_hud_client,
+        data_source: hmis_ds,
+        RaceNone: nil,
+        GenderNone: nil,
+        **race_fields.map { |field| [field, 99] }.to_h,
+        **gender_fields.map { |field| [field, 99] }.to_h,
+        AmIndAKNative: 1,
+        Woman: 1,
+      )
+    end
+
+    let!(:c2_doesnt_know) do
+      create(
+        :hmis_hud_client,
+        data_source: hmis_ds,
+        RaceNone: 8,
+        GenderNone: 8,
+        **race_fields.map { |field| [field, 99] }.to_h,
+        **gender_fields.map { |field| [field, 99] }.to_h,
+      )
+    end
+
+    let!(:c3_prefers_not_to) do
+      create(
+        :hmis_hud_client,
+        data_source: hmis_ds,
+        RaceNone: 9,
+        GenderNone: 9,
+        **race_fields.map { |field| [field, 99] }.to_h,
+        **gender_fields.map { |field| [field, 99] }.to_h,
+      )
+    end
+
+    let!(:c4_data_not_collected) do
+      create(
+        :hmis_hud_client,
+        data_source: hmis_ds,
+        RaceNone: 99,
+        GenderNone: 99,
+        **race_fields.map { |field| [field, 99] }.to_h,
+        **gender_fields.map { |field| [field, 99] }.to_h,
+      )
+    end
+
+    let!(:c5_none_field_nil) do
+      create(
+        :hmis_hud_client,
+        data_source: hmis_ds,
+        RaceNone: nil,
+        GenderNone: nil,
+        **race_fields.map { |field| [field, 0] }.to_h,
+        **gender_fields.map { |field| [field, 0] }.to_h,
+      )
+    end
+
+    let!(:other_ds_client) do
+      create(
+        :hmis_hud_client,
+        data_source: other_source_ds,
+        RaceNone: 99,
+        GenderNone: 99,
+        **race_fields.map { |field| [field, 99] }.to_h,
+        **gender_fields.map { |field| [field, 99] }.to_h,
+      )
+    end
+
+    it 'updates race/gender values of 99 to 0' do
```

Comment:
This seems like it would be well suited to the test style
```
expect do 
   HmisDataCleanup::Util.fix_race_gender_99s!
end.to change { client.reload.field }.from(x).to(y).and
  not_change { ... }.and
  etc
```

Since it looks like you're asserting on some fields that _weren't_ supposed to change, and some fields that _were_. It makes it a little easier to read the test because you can see which fields were expected to change.

Reply:
Ok, it is quite a bit more verbose now, but hopefully clearer as well. Let me know if you think it looks okay.